psi4/src/psi4/libmints/oeprop.cc

```diff
@@ -1784,6 +1788,445 @@ PopulationAnalysisCalc::compute_lowdin_charges(bool print_output) {
+                    octupoles[a][k][i][j] = octupoles[a][i][j][k];
+                    octupoles[a][k][j][i] = octupoles[a][i][j][k];
+
+                    stone_octupoles[a][j][i][k] = stone_octupoles[a][i][j][k];
+                    stone_octupoles[a][j][k][i] = stone_octupoles[a][i][j][k];
+                    stone_octupoles[a][i][k][j] = stone_octupoles[a][i][j][k];
+                    stone_octupoles[a][k][i][j] = stone_octupoles[a][i][j][k];
+                    stone_octupoles[a][k][j][i] = stone_octupoles[a][i][j][k];
+                }
+            }
+        }
+    }
+
+    // non-redundant cartesian
+    auto mpole_flat = std::make_shared<Matrix>("MBIS Charges: (a.u.)", num_atoms, 1);
+    auto dpole_flat = std::make_shared<Matrix>("MBIS Dipoles: (a.u.)", num_atoms, 3);
+    auto qpole_flat = std::make_shared<Matrix>("MBIS Quadrupoles: (a.u.)", num_atoms, 6);
+    auto opole_flat = std::make_shared<Matrix>("MBIS Octupoles: (a.u.)", num_atoms, 10);
+
+    for(int a = 0; a < num_atoms; a++) {
+        mpole_flat->set(a, 0, monopoles[a]);
+
+        dpole_flat->set(a, 0, dipoles[a][0]);
+        dpole_flat->set(a, 1, dipoles[a][1]);
+        dpole_flat->set(a, 2, dipoles[a][2]);
+
+        qpole_flat->set(a, 0, quadrupoles[a][0][0]);
+        qpole_flat->set(a, 1, quadrupoles[a][0][1]);
+        qpole_flat->set(a, 2, quadrupoles[a][0][2]);
+        qpole_flat->set(a, 3, quadrupoles[a][1][1]);
+        qpole_flat->set(a, 4, quadrupoles[a][1][2]);
+        qpole_flat->set(a, 5, quadrupoles[a][2][2]);
+
+        opole_flat->set(a, 0, octupoles[a][0][0][0]);
+        opole_flat->set(a, 1, octupoles[a][0][0][1]);
+        opole_flat->set(a, 2, octupoles[a][0][0][2]);
+        opole_flat->set(a, 3, octupoles[a][0][1][1]);
+        opole_flat->set(a, 4, octupoles[a][0][1][2]);
+        opole_flat->set(a, 5, octupoles[a][0][2][2]);
+        opole_flat->set(a, 6, octupoles[a][1][1][1]);
+        opole_flat->set(a, 7, octupoles[a][1][1][2]);
+        opole_flat->set(a, 8, octupoles[a][1][2][2]);
+        opole_flat->set(a, 9, octupoles[a][2][2][2]);
+    }
+
+    if (print_output) {
+
+        outfile->Printf("  MBIS Charges: (a.u.)\n");
+        outfile->Printf("   Center  Symbol  Z      Pop.       Charge\n");
+
+        for (int a = 0; a < num_atoms; a++) {
+            outfile->Printf("  %5d      %2s %4d   %9.6f   %9.6f\n", a+1, mol->label(a).c_str(), 
+                    (int)mol->Z(a), mol->Z(a) - monopoles[a], monopoles[a]);
+        }
+
+        outfile->Printf("\n  MBIS Dipoles: (a.u.)\n");
+        outfile->Printf("   Center  Symbol  Z        X           Y           Z\n");
+
+        for (int a = 0; a < num_atoms; a++) {
+            outfile->Printf("  %5d      %2s %4d   %9.6f   %9.6f   %9.6f\n", a+1, mol->label(a).c_str(), 
+                    (int)mol->Z(a), dpole_flat->get(a, 0), dpole_flat->get(a, 1), dpole_flat->get(a, 2));
+        }
+
+        outfile->Printf("\n  MBIS Quadrupoles: (a.u.)\n");
```

Comment:
good to label units specifically: `[e a0]` for Dipole, `[e a0^2]` for Quad, etc.